src/main/java/com/coveo/k8sproxy/proxy/K8sReverseProxy.java

```diff
@@ -78,6 +79,47 @@
     private Timer timer = new Timer();
     private ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
 
+    @RequestMapping(value = "/k8s_cluster_endpoint", method = RequestMethod.GET)
+    @ResponseBody
+    public ClusterEndpoint getCurrentEndpoint(HttpServletRequest request, HttpServletResponse response)
+    {
+        return new ClusterEndpoint(k8sClusterEndpoint);
+    }
+
+    @RequestMapping(value = "/k8s_cluster_endpoint/set", method = RequestMethod.GET)
+    @ResponseBody
+    public ClusterEndpoint setCurentEndpoint(@RequestParam String endpoint,
+                                             HttpServletRequest request,
+                                             HttpServletResponse response)
+            throws MissingParameterException,
+                InvalidParameterException,
+                IOException
+    {
+        if (endpoint == null) {
```

Comment:
`endpoint` cannot be null if you didn't specify `@RequestParam` as optional (it can be an empty string though but that should be caught by your other validation)